File: src/main/java/io/lsdconsulting/lsd/distributed/generator/diagram/dto/EventContainer.java
package io.lsdconsulting.lsd.distributed.generator.diagram.dto;

import com.lsd.core.domain.SequenceEvent;
import lombok.Builder;
import lombok.Value;

import java.time.ZonedDateTime;
import java.util.List;

@Value
@Builder
public class EventContainer {
    List<SequenceEvent> events;
    ZonedDateTime startTime;
    ZonedDateTime finishTime;
}

File: src/main/kotlin/io/lsdconsulting/lsd/distributed/generator/diagram/dto/EventContainer.kt
package io.lsdconsulting.lsd.distributed.generator.diagram.dto

import com.lsd.core.domain.SequenceEvent
import java.time.ZonedDateTime

data class EventContainer(
    var events: List<SequenceEvent>,
    var startTime: ZonedDateTime? = null,
    var finishTime: ZonedDateTime? = null,
)